### DISCUSSION TITLE
Poetry: Inconsistent Updating of Dependencies in pyproject.toml

### DISCUSSION THREAD

**Discussion by Piwero** [renovatebot/renovate#25960] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

None

### The Issue:
It seems like Renovate is only updating the poetry.lock file and not consistently updating the corresponding entries in the pyproject.toml file. For instance, in [this PR](https://github.com/Piwero/crud_auth_github/commit/a7eeeef94573edc593eefb43ceef0b28af4d8a03), the dependency mypy got updated in poetry.lock but not in pyproject.toml.

### Previous Behavior:
I noticed that in [an earlier PR](https://github.com/Piwero/crud_auth_github/commit/a62d9ad231d0854392248cf4797b60e02a5863ca), Renovate correctly updated both files for the ruff dependency.

### Configuration Check:
I double-checked my Renovate configuration, and everything seems to be in order. The bot should be updating both files, as it did in the past.

### Request for Assistance:
Any insights or suggestions on why this might be happening? I'm a bit stumped, and your guidance would be greatly appreciated.

Thanks a bunch for your time and your fantastic work on Renovate!

### Relevant debug logs

<details><summary>Logs</summary>

```



```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment, marked as answer]
There is no problem with the PR you linked to and it is a valid update. By default, Renovate only updates ranges in the package file if the new version no longer satisfies that range.

I suggest you just pin your dependencies to exact versions in the pyproject.toml and then every update will update both the package file and lock file.

Read up the docs for `rangeStrategy` for more details

**Reply by Piwero** [reply]
Alright, I appreciate this. I assumed this was already in place, which is why I was only able to see it in the most recent PR. Thank you for directing me to the documentation section for more information. I'll look into this.
